Docs/poc/pg4kedbrepohappypath (#6169)
EnterpriseDB/docs: product_docs/docs/postgres_for_kubernetes/1/installation_upgrade.mdx

## Installation on Kubernetes

### Obtaining an EDB subscription token

Installing EDB Postgres for Kubernetes requires an EDB Repos 2.0 token to gain access to the EDB private software repositories. 

> You can obtain the token by visiting the [EDB Repos website](https://www.enterprisedb.com/repos) and signing into your account.

Author: ```suggestion
You can obtain the token by visiting your [EDB Repos account page](https://www.enterprisedb.com/accounts/progile) and signing into your account.
```

Your account profile page displays the token to use - by default it is obscured, click the "Show" button (an eye icon) to reveal it.

Your token entitles you to access one of two repositories: standard or enterprise. 

> * Standard - Includes the operator and operands for EDB Postgres Extended.

Author: ```suggestion
* Standard - Includes the operator and standard operands, including EDB Postgres Extended.
```

> * Enterprise - Includes the operator and operands for EDB Postgres Advanced and EDB Postgres Extended.

Reviewer: both include the standard operands, as well as the operators. Worth mentioning?

Author: Applied a tweak to the text.

> * Enterprise - Includes the operator and operands for EDB Postgres Advanced and EDB Postgres Extended.

Author: ```suggestion
* Enterprise - Includes the operator and standard and enterprise operands, including EDB Postgres Advanced and EDB Postgres Extended.
```

> 
> Export the appropriate value as an environmental variable EDB_SUBSCRIPTION_PLAN.
> 

Reviewer 2: ```suggestion

Export the appropriate value as an environment variable EDB_SUBSCRIPTION_PLAN.

```

```shell

> export EDB_SUBSCRIPTION_PLAN=enterprise

Reviewer 2: You don't need to export the variables to use them in the same shell.

```suggestion
EDB_SUBSCRIPTION_PLAN=enterprise
```

```

then export the Repos 2.0 token to an environment variable EDB_SUBSCRIPTION_TOKEN.

```shell

> export EDB_SUBSCRIPTION_TOKEN=<your-token>

Reviewer 2: You don't need to export the variables to use them in the same shell.

Suggested change
```suggestion
EDB_SUBSCRIPTION_TOKEN=<your-token>
```

```

!!! Warning
The token is sensitive information. Please ensure that you don't expose it to unauthorized users.
!!!

### Using the Helm Chart

The operator can be installed using the provided [Helm chart](https://github.com/EnterpriseDB/edb-postgres-for-kubernetes-charts).

### Directly using the operator manifest

#### Install the EDB pull secret

> Before installing EDB Postgres for Kubernetes, you need to create a pull secret for EDB software in the `postgresql-operator-system` namespace.

Reviewer: I know you already mention "EDB software" in the section on getting the token, but I think this paragraph is clearer if saying the pull secret gives access to the private registries. I.e. not to "EDB software" in general.

Author: ```suggestion
Before installing EDB Postgres for Kubernetes, you need to create a pull secret for EDB private registries in the `postgresql-operator-system` namespace.
```

> The pull secret needs to be save in the namespace where the operator will reside. Create a postgresql-operator-system namespace using this command:
> 

Reviewer 2: ```suggestion
The pull secret needs to be saved in the namespace where the operator will reside. Create a postgresql-operator-system namespace using this command:

```

```shell
kubectl create namespace postgresql-operator-system
```

To create the pull secret itself, run the following command:

```shell
kubectl create secret -n postgresql-operator-system docker-registry edb-pull-secret \
  --docker-server=docker.enterprisedb.com \

>   --docker-username=k8s_$EDB_SUBSCRIPTION_PLAN \
>   --docker-password=$EDB_SUBSCRIPTION_TOKEN

Reviewer 2: ```suggestion
  --docker-username="k8s_${EDB_SUBSCRIPTION_PLAN}" \
  --docker-password="${EDB_SUBSCRIPTION_TOKEN}"
```

```

#### Install the operator

The operator can be installed like any other resource in Kubernetes,
through a YAML manifest applied via `kubectl`. There are two different manifests available depending on your subscription plan:

- Standard: The [latest standard operator manifest](https://get.enterprisedb.io/pg4k/pg4k-standard-1.24.1.yaml).
- Enterprise: The [latest enterprise operator manifest](https://get.enterprisedb.io/pg4k/pg4k-enterprise-1.24.1.yaml).

You can install the manifest for the latest version of the operator by running:

```sh
kubectl apply --server-side -f \

>   https://get.enterprisedb.io/pg4k/pg4k-enterprise-1.24.1.yaml
> 

Reviewer 2: ```suggestion
  "https://get.enterprisedb.io/pg4k/pg4k-${EDB_SUBSCRIPTION_PLAN}-1.24.1.yaml"

```
